js/msp.js

```diff
@@ -109,14 +109,18 @@ var MSP = {
                     }
                     break;
                 case 9:
-                    if (this.message_checksum == data[i]) {
-                        // message received, store dataview
-                        this.dataView = new DataView(this.message_buffer, 0, this.message_length_expected);
-                    } else {
+                    if (this.message_checksum != data[i]) {
                         console.log('code: ' + this.code + ' - crc failed');
-                        this.dataView = null;
                         this.packet_error++;
+                        
+                        // if this is a dataflash read packet then use
+                        // the compressionType byte as a flag to indicate
+                        // that a crc error occurred and the packet should retry
+                        if (this.code == MSPCodes.MSP_DATAFLASH_READ) {
+                            this.message_buffer_uint8_view[6] = 255;
```

Comment:
It's looking like this may be beyond my skills.  I can't find where the `response` object that is passed back to the callback from `MSP.send_message` is defined.  It looks like we would need to add the `CrcError` flag to the response object but maybe I'm not understanding.  And then how does that get populated from method read in msp.js?  Just not understanding the code structure at this point.